tests/unit/roster/terraform.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+'''
+unittests for terraform roster
+'''
+from __future__ import absolute_import
+import os.path
+from salt.roster import terraform
+
+from tests.support.unit import TestCase
+
+
+class TerraformTestCase(TestCase):
+    '''
+    Test cases for salt.roster.terraform
+    '''
+    def setUp(self):
+        terraform.__opts__ = {}
+
+    def test_default_output(self):
+        '''
+        Test the output of a fixture tfstate file wich contains libvirt
+        and AWS resources.
+        '''
+        tfstate = os.path.join(os.path.dirname(__file__), 'terraform.data', 'terraform.tfstate')
+        terraform.__opts__['roster_file'] = tfstate
+        pki_dir = os.path.abspath(
+            os.path.join(os.path.dirname(__file__), 'terraform.data'))
+        terraform.__opts__['pki_dir'] = pki_dir
+
+        expected_result = {
+            'minsles12sp1': {
+                'host': '192.168.127.222',
+                'priv': os.path.join(pki_dir, 'ssh/salt-ssh.rsa')},
+            'package-mirror': {
+                'host': '192.168.127.157',
+                'priv': os.path.join(pki_dir, 'ssh/salt-ssh.rsa')},
+            'moio-sumaform-package-mirror': {
+                'host': 'ec2-54-209-95-95.compute-1.amazonaws.com',
+                'priv': os.path.join(pki_dir, 'ssh/salt-ssh.rsa')},
+            'suma3pg': {
+                'host': '192.168.127.187',
+                'priv': os.path.join(pki_dir, 'ssh/salt-ssh.rsa')}
+        }
+
+        ret = terraform.targets('*')
+        self.assertDictEqual(expected_result, ret)
+
+
+if __name__ == '__main__':
+    import unittest
+    unittest.main()
```

Comment:
Pytest works if you run a single test module or only the integration tests or probably only the unittests(but these will fail for sure). Once you mix unittest with integration tests it will fail badly. The reason is that salt's dunders are being populated at the module level instead of being patched by mock on `setUp` which has all kinds of secondary effects.

However, pytest is the goal, but a lot of cleaning up must happen before we completely move to pytest, unfortunately.